arrow/ipc/file_reader.go

```diff
@@ -768,3 +867,44 @@ func readDictionary(memo *dictutils.Memo, meta *memory.Buffer, body ReadAtSeeker
 	}
 	return dictutils.KindReplacement, nil
 }
+
+type mappedFileBlock struct {
+	offset int64
+	meta   int32
+	body   int64
+
+	data []byte
+}
+
+func (blk mappedFileBlock) Offset() int64 { return blk.offset }
+func (blk mappedFileBlock) Meta() int32   { return blk.meta }
+func (blk mappedFileBlock) Body() int64   { return blk.body }
+
+func (blk mappedFileBlock) section() []byte {
+	return blk.data[blk.offset : blk.offset+int64(blk.meta)+blk.body]
+}
+
+func (blk mappedFileBlock) NewMessage() (*Message, error) {
+	var (
+		body *memory.Buffer
+		meta *memory.Buffer
+		buf  = blk.section()
+	)
+
+	metaBytes := buf[:blk.meta]
+
+	prefix := 0
+	switch binary.LittleEndian.Uint32(metaBytes) {
```

Comment:
int32 not uint32?

https://github.com/apache/arrow/blob/313d11aa94c2be71142b55e3d8bb166d780c19c7/format/File.fbs#L45

> metaDataLength: int;